### PR TITLE
Update speech translation task and dataset

### PR DESCRIPTION
**Pull request by xutaima**
# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
- [ ] Did you read the [contributor guideline](https://github.com/pytorch/fairseq/blob/master/CONTRIBUTING.md)?
- [ ] Did you make sure to update the docs?   
- [ ] Did you write any new necessary tests?  

## What does this PR do?
Fixes # (issue).
- Refactor the code for shared task
- Add options for online and offline feature extraction
- Add options to disable feature normalization along the time axis

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: examples/simultaneous_translation/criterions/cross_entropy_acc.py [hunk 1]
```diff
@@ -13,122 +13,6 @@
 from fairseq import utils
 from fairseq.criterions import FairseqCriterion, register_criterion
 
-
-@register_criterion("cross_entropy_acc")
-class CrossEntropyWithAccCriterion(FairseqCriterion):
```
Comment (jmp84) [inline examples/simultaneous_translation/criterions/cross_entropy_acc.py]: Why remove this?

File: examples/simultaneous_translation/data/ast_dataset.py [hunk 2]
```diff
@@ -64,47 +56,33 @@ def __getitem__(self, index):
         path = self.aud_paths[index]
         if not os.path.exists(path):
             raise FileNotFoundError("Audio file not found: {}".format(path))
```
Comment (jmp84) [inline examples/simultaneous_translation/data/ast_dataset.py]: Note that is your features and wav files are not side by side, this will raise an error even though the feature file exists.

File: examples/simultaneous_translation/data/ast_dataset.py [hunk 3]
```diff
@@ -64,47 +56,33 @@ def __getitem__(self, index):
...
+            output = torch.load(
+                f'{path}.{self.num_mel_bins}-{self.frame_length}-{self.frame_shift}{"-use_energy"*self.use_energy}.fbank'
+            )
+            
+        if self.use_energy:
+            energy, output = output[:, 0], output[:, 1:]
+        else:
+            energy = None
+
+        import pdb; pdb.set_trace()
```
Comment (jmp84) [inline examples/simultaneous_translation/data/ast_dataset.py]: remove this

File: examples/simultaneous_translation/docs/baseline.md [hunk 4]
```diff
@@ -134,7 +134,8 @@ CUDA_VISIBLE_DEVICES=1 python $FAIRSEQ/train.py data-bin/mustc_en_de \
     --log-format json \
     --log-interval 10 \
     --criterion cross_entropy_acc \
-    --user-dir $FAIRSEQ/examples/simultaneous_translation
+    --user-dir $FAIRSEQ/examples/simultaneous_translation \
+    --online-features
```
Comment (jmp84) [inline examples/simultaneous_translation/docs/baseline.md]: What's the default?

File: examples/simultaneous_translation/criterions/cross_entropy_acc.py (hunk 1 above)
Comment (xutaima) [reply examples/simultaneous_translation/criterions/cross_entropy_acc.py]: Now I import this from speech recognition repo so this will raise a duplicated registration error

File: examples/simultaneous_translation/docs/baseline.md (hunk 4 above)
Comment (xutaima) [reply examples/simultaneous_translation/docs/baseline.md]: The default is to read offline features from disk, this flag will do online extraction. Or is it better to have offline as default?

File: examples/simultaneous_translation/data/ast_dataset.py (hunk 2 above)
Comment (jmp84) [reply examples/simultaneous_translation/data/ast_dataset.py]: Any comment on this one?

File: examples/simultaneous_translation/docs/baseline.md (hunk 4 above)
Comment (jmp84) [reply examples/simultaneous_translation/docs/baseline.md]: If offline is faster, it should be a good default.
